lightning/src/ln/outbound_payment.rs

```diff
@@ -1176,6 +1178,7 @@ impl OutboundPayments {
 						full_failure_ev = Some(events::Event::PaymentFailed {
 							payment_id: *payment_id,
 							payment_hash: payment.get().payment_hash().expect("PendingOutboundPayments::RetriesExceeded always has a payment hash set"),
+							reason: PaymentFailureReason::RecipientRejected
```

Comment:
For `PaymentFailureReason::RecipientRejected` here, it seems difficult to distinguish whether it’s actually because the recipient rejected the payment or because the payment failed somewhere along the path (will probably change the name to reflect this), not sure if this is worth trying to do or if it can be easily done since from my understanding of onion error messages we’d probably have to look at the actual error data?

Reply:
Oh nice!

Reply:
Added this, along with `FailedAlongPath` variant, and used similar conditions (relating to failing permanently or being retryable) in tests to accomodate for this change